stacks/cli/src/interface.ts

```diff
@@ -18,3 +18,54 @@ export abstract class Command extends BaseCommand<
   CommandMetadataInput,
   CommandMetadataOption
 > {}
+
+export interface Function {
+  _id?: string;
+  name?: string;
+  description?: string;
+  env?: Environment;
+  triggers: Triggers;
+  memoryLimit?: number;
+  timeout?: number;
+  indexPath?: string;
+  dependencies?: string;
+}
+
+export interface Triggers {
+  default: Trigger;
+  [key: string]: Trigger;
+}
+
+export interface Trigger {
+  type: string;
+  active?: boolean;
+  options: any;
+}
+
+export interface Environment {
+  [key: string]: string;
+}
+
+export interface Dependency {
+  [key: string]: string;
+}
+
+export interface Asset {
+  kind: string;
+  metadata: MetaData;
+  spec: Function | any;
```

Comment:
@htuna07 Doesn't this line mean spec is always "any"?

Reply:
But I didn't know how to define this property before we decide asset types like **Bucket**, **Bucket Data** , **Function**..

Reply:
But yes, it means always "any" like you said. What should we do then ?

Reply:
@htuna07 We should use the types if we've them. You might use Any for now but create a tech story for us to discuss later.